File: src/main/kotlin/com/leetcode/P2740.kt
package com.leetcode

// https://github.com/antop-dev/algorithm/issues/567
class P2740 {
    fun findValueOfPartition(nums: IntArray): Int {
        nums.sort()
        var ans = 1_000_000_000
        for (i in 0 until nums.size - 1) {
            val diff = nums[i + 1] - nums[i]
            ans = minOf(ans, diff)
        }
        return ans
    }
}

File: src/python3/com/leetcode/p2740.py
class Solution(object):
    def findValueOfPartition(self, nums):
        nums.sort()
        ans = 1000000000  # 1_000_000_000
        for i in range(len(nums) - 1):
            diff = nums[i + 1] - nums[i]
            ans = min(ans, diff)
        return ans
